libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -335,6 +327,7 @@ AP_GPS::detect_instance(uint8_t instance)
 found_gps:
 #endif
 	if (new_gps != NULL) {
+        _broadcast_gps_type(instance);
```

Comment:
Shouldn't this be the last thing inside the if? I mean, the `drivers[instance]` is set after this but used inside `_detection_message`.